orange/src/test/java/com/hartwig/hmftools/orange/algo/purple/PurpleVariantFactoryTest.java

```diff
@@ -28,135 +23,102 @@
 import com.hartwig.hmftools.orange.algo.pave.PaveAlgo;
 import com.hartwig.hmftools.orange.algo.pave.TestEnsemblDataCacheFactory;
 
-import org.apache.logging.log4j.util.Strings;
-import org.jetbrains.annotations.NotNull;
 import org.junit.Test;
 
-public class PurpleVariantFactoryTest {
-
-    private static final double EPSILON = 1.0E-10;
+public class PurpleVariantFactoryTest
+{
 
     @Test
-    public void canHandleNullVariants() {
-        PurpleVariantFactory factory = createTestFactory();
-        assertNull(factory.create(null));
-    }
+    public void testCanCreateVariantFromContext()
+    {
+        var otherImpacts = List.of(
+                new VariantTranscriptImpact(
+                        "[ENSG00000109265",
```

Comment:
I think your logic should filter out these brackets at an early stage. I don't think your own `PurpleVariantContext` should ever contain brackets so you also don't have to test the conversion with brackets. If your `PurpleVariantContext` can contains brackets, you should deal with those in the loader.

Reply:
Good suggestion and I agree. The inclusion of the brackets does seem unintentional and in my opinion it ideally should be fixed in the `VariantTranscriptImpact` class itself. Maybe I can discuss this with Charles some time.